### PR TITLE
MUL-1029: BTC tx to P2SH address

### PR DESCRIPTION
**Pull request by PashaKlybik**
-make transaction with PubScript <P2SH> destination
 -make transaction with PubScript <P2SH-segwit-addresses> destination
Also task MUL-1066: add tests with P2SH addresses testnet/mainne

### CODE REVIEW

**Comment by PashaKlybik** [comment]
@Enmk

File: multy_core/src/api/account.cpp [hunk 1]
```diff
@@ -230,9 +230,9 @@ Error* validate_address(BlockchainType blockchain_type, const char* address)
                 BitcoinAddressType address_type;
                 BitcoinNetType net_type;
                 bitcoin_parse_address(address, &net_type, &address_type);
-                if (address_type != BITCOIN_ADDRESS_P2PKH)
+                if (address_type != BITCOIN_ADDRESS_P2PKH && address_type != BITCOIN_ADDRESS_P2SH)
                 {
-                    THROW_EXCEPTION("BTC: Only P2PKH addresses are supported for now.");
+                    THROW_EXCEPTION("BTC: Only P2PKH and P2SH addresses are supported for now.");
```
Comment (Enmk) [inline multy_core/src/api/account.cpp]: Please update the message:
```
THROW_EXCEPTION("Not supported address type, only P2PKH and P2SH are supported.") << " Address type: " << address_type;
```

File: multy_test/test_bitcoin_transaction.cpp [hunk 2]
```diff
@@ -1118,3 +1118,100 @@ GTEST_TEST(BitcoinTransactionTest, SmokeTest_mainnet_with_op_return)
                 "beb234dba313a9f0928fb08ab7ca086a74a1ae18da0e663c0aac0ffffffff020000000000000000106a0e4d554c545920746865"
                 "206265737419a10100000000001976a91441d29eaed90c0a26021daa5139dcc1d45c4e34a188ac00000000")), *serialied);
 }
+
+GTEST_TEST(BitcoinTransactionTest, SmokeTest_testnet_to_P2SH_addres)
+{
+    AccountPtr account;
+    HANDLE_ERROR(make_account(
```
Comment (Enmk) [inline multy_test/test_bitcoin_transaction.cpp]:
```
        AccountPtr account = make_account(BLOCKCHAIN_BITCOIN,
                "cQeGKosJjWPn9GkB7QmvmotmBbVg1hm8UjdN6yLXEWZ5HAcRwam7");
```

File: multy_test/test_bitcoin_transaction.cpp [hunk 3]
```diff
@@ -1118,3 +1118,100 @@ GTEST_TEST(BitcoinTransactionTest, SmokeTest_mainnet_with_op_return)
...
+        },
+        { // Destinations
+            TransactionDestination
+            {
+                "2MwsRtyuZyW2HuPUcUKGa4dAm9M9RkUPacK",
+                sent
+            }
+        }
+    };
+    TransactionPtr transaction = make_transaction_from_template(TEST_TX);
```
Comment (Enmk) [inline multy_test/test_bitcoin_transaction.cpp]:
```
        transaction = make_transaction_from_template(temp, TEST_TX,
                account->get_private_key());
```

File: multy_test/test_bitcoin_transaction.cpp [hunk 4]
```diff
@@ -1118,3 +1118,100 @@ GTEST_TEST(BitcoinTransactionTest, SmokeTest_mainnet_with_op_return)
...
+    const BinaryDataPtr serialied = transaction->serialize();
+
+    // Txid: c8a8fe4e36d006e28dc55f00bbc7cfea9622b6262710973f6625f4fb9710cba4
+    ASSERT_EQ(as_binary_data(from_hex(
+                "01000000017ccb19731651a9acfd48a8810211127f42c1e45391474ca9693da6af4e70f282020000008b4830450221009c526a9"
+                "d76f125df9b874d7803b6f72c7c116ee847338771c68919a189e666fb02203a6f6ebea415561d9846816c02d0a93596ddb92ae3"
+                "57436a36745c02c92bff2f014104a7364b8ad3f02ae0a9dbe355aa3ac9b2a661e230976f505f55f66d26ea3ae906ed3bbfd24ab"
+                "384fe7c15a99ae1227276629820da47f10c29fd9c9a026bc602deffffffff0144bf01000000000017a914aaf44ab6c1076eae84"
+                "9bf717adf955e0456639ab8700000000")), *serialied);
+}
```
Comment (Enmk) [inline multy_test/test_bitcoin_transaction.cpp:1207]: Please add a test that verifies destination address validation.
2 cases when validation should pass: 
* P2PKH 
* P2SH
and one case where it fails:
* some bogus address.

**Review by Enmk — CHANGES_REQUESTED** [review]
Minor fixes + add more test cases

File: multy_core/src/bitcoin/bitcoin_transaction.cpp [hunk 5]
```diff
@@ -41,6 +41,17 @@ uint32_t bitcoin_traits()
...
 }
 
+enum OP_CODE : uint8_t
+{
+    OP_FALSE = 0x00,
+    OP_RETURN = 0x6A,
+    OP_DUP = 0x76,
+    OP_HASH160 = 0xA9,
+    OP_EQUALVERIFY = 0x88,
+    OP_CHECKSIG =0xAC,
```
Comment (Enmk) [inline multy_core/src/bitcoin/bitcoin_transaction.cpp]: ```OP_CHECKSIG =<MISSING SPACE HERE>0xAC,```

**Review by Enmk — APPROVED** [review]
Ok

**Review by Enmk — COMMENTED** [review]
Ok

**Pull request merged**